packages/sirius-web/frontend/sirius-web-application/src/omnibox/PublishLibraryDialog.tsx

```diff
@@ -0,0 +1,122 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+import DialogContentText from '@mui/material/DialogContentText';
+import DialogContent from '@mui/material/DialogContent';
+import DialogTitle from '@mui/material/DialogTitle';
+import DialogActions from '@mui/material/DialogActions';
+import Button from '@mui/material/Button';
+import TextField from '@mui/material/TextField';
+import { makeStyles } from 'tss-react/mui';
+import { useState } from 'react';
+import { usePublishLibraries } from './usePublishLibraries';
+import { useCurrentProject } from '../views/edit-project/useCurrentProject';
+import Link from '@mui/material/Link';
+import Dialog from '@mui/material/Dialog';
+import { PublishLibraryDialogProps, PublishLibraryDialogState } from './PublishLibraryDialog.types';
+import { Link as RouterLink } from 'react-router-dom';
+
+const usePublishCommandStyles = makeStyles()((theme) => ({
+  form: {
+    display: 'flex',
+    flexDirection: 'column',
+    '& > *': {
+      marginBottom: theme.spacing(1),
+    },
+  },
+  link: {
+    fontStyle: 'italic',
+  },
+}));
+
+export const PublishLibraryDialog = ({ open, title, message, publicationKind, onClose }: PublishLibraryDialogProps) => {
+  const { classes } = usePublishCommandStyles();
+
+  const [state, setState] = useState<PublishLibraryDialogState>({
+    version: '',
+    versionIsInvalid: true,
+    pristine: true,
+    description: '',
+  });
+
+  const onVersionChange: React.ChangeEventHandler<HTMLInputElement | HTMLTextAreaElement> = (event) => {
+    const value = event.target.value;
+    const versionIsInvalid = value.trim().length === 0;
```

Comment:
You don't need that in the state, it can easily be derived from the state while rendering.

Reply:
Done